pkg/csidriver/driver.go

```diff
@@ -547,19 +551,27 @@ func mapBackingResourceToPod(dv *driverVolume) error {
 		cmNamespace := sharedConfigMap.Spec.ConfigMapRef.Namespace
 		cmName := sharedConfigMap.Spec.ConfigMapRef.Name
 		comboKey := objcache.BuildKey(cmNamespace, cmName)
-		cm := client.GetConfigMap(cmNamespace, cmName)
-		if cm != nil {
-			payload := Payload{
-				StringData: cm.Data,
-				ByteData:   cm.BinaryData,
-			}
 
-			upsertError := commonUpsertRanger(dv, comboKey, payload)
-			if upsertError != nil {
-				ProcessFileSystemError(cm, upsertError)
-				return upsertError
+		cm, err := client.GetConfigMap(cmNamespace, cmName)
+		if err != nil { // If the error is `Permission Denied, it returns a `PermissionDenied` error, which correctly fails the initial volume mount
+			if kerrors.IsForbidden(err) {
+				return status.Errorf(codes.PermissionDenied, "CSI driver does not have permission to get configmap %s/%s", cmNamespace, cmName)
 			}
+			klog.Warningf("could not retrieve configmap %s/%s: %v", cmNamespace, cmName, err)
+			return nil
```

Comment:
Are these change in execution required for the e2e test?
I feel these changes should be captured in a different PR

Reply:
Maybe the need here was the kubelet needs those GRPC error codes vs. what is in the k8s forbidden exception?

If so, then I could see how you would want to return permission denied here 

but I would not reutrn nil if it is a non permission error .... either return the original error like before, or find the generic GRPC error code and do a status.Errorf... with that